test/e2e/storage/volume_metrics.go

```diff
@@ -113,7 +113,11 @@ var _ = SIGDescribe("[Serial] Volume metrics", func() {
 				return false, err
 			}
 			updatedStorageMetrics = getControllerStorageMetrics(updatedMetrics)
-			if len(updatedStorageMetrics) == 0 {
+			metricCount := len(updatedStorageMetrics)
+			// Usually a pod deletion does not mean immediate volume detach
+			// we will have to retry to verify volume_detach metrics
+			_, detachMetricFound := updatedStorageMetrics["volume_detach"]
+			if metricCount < 3 || !detachMetricFound {
```

Comment:
Is the number 3 it obvious? Not to me but I am not knowledgeable in metrics. Perhaps this should be a descriptive constant?

Reply:
in test below I am measuring 3 metrics and hence I expect at least 3 metrics to be present in the result. But because `volume_detach` metric will be generated at very last, i am separately checking that in case code generates other metrics. 

So 3 isn't really a magic number - just number of metrics test is checking. i can document it.